### DISCUSSION TITLE
How does implicit messaging work?

### DISCUSSION THREAD

**Discussion by YairLevi** [EIPStackGroup/OpENer#515] [Q&A]
Hi, 
I'm looking for someone to answer some questions I have regarding implicit messaging in EIP.
I can't find any official forum or anything so I'm trying to get help here.

In implicit messaging, I never see any code example of someone defining the exact data (instance/class) that they wish to receive, all I see is specifying some pre defined assmebly object.

Can anyone explain to me how this works? where can i define what I want to receive in implicit messaging? and how do I deduce from the response the values?

**Comment by azoitl** [comment]
in CIP the data that is received or sent via implicit messaging are always an assembly object. An assembly object takes data from different objects and provides it in form of a byte array. How the data is assembled into the assembly object is either defined by certain profiles or vendor specific. 

In the [ODVA library](https://www.odva.org/technology-standards/document-library/) there are several documents that explain the basic workings of CIP and EIP. I personally can recommend pubs 122, 123, and 138 to get you started.
